src/Core/Framework/DataAbstractionLayer/EntityRepository.php

```diff
@@ -95,7 +95,7 @@ public function search(Criteria $criteria, Context $context): EntitySearchResult
 
         $ids = $this->searchIds($criteria, $context);
 
-        $readCriteria = $criteria->cloneForRead($ids->getIds());
+        $readCriteria = $criteria->cloneForRead(empty($ids->getIds()) ? null : $ids->getIds());
```

Comment:
No read should happen if there are no ids.

Reply:
No search happening, I hope I implemented a good alternative